packages/theme/pages/MyAccount/SavedAddressesDetails.vue

```diff
@@ -0,0 +1,213 @@
+              <SfIcon
+                icon="cross"
+                color="gray"
+                size="14px"
+                role="button"
+                class="smartphone-only"
+                @click="removeAddress(address)"
+              />
+              <SfButton
+                @click="changeAddress(address)">
+                {{ $t('Change') }}
+              </SfButton>
+
+              <SfButton
+                class="color-light savedAddresses__button-delete desktop-only"
+                @click="removeAddress(address)">
+                {{ $t('Delete') }}
+              </SfButton>
+            </div>
+          </div>
+        </transition-group>
+        <SfButton
+          class="action-button"
+          @click="changeAddress()">
+          {{ $t('Add new address') }}
+        </SfButton>
+      </SfTab>
+    </SfTabs>
+  </transition>
+</template>
+<script>
+import {
+  SfTabs,
+  SfButton,
+  SfIcon
+} from '@storefront-ui/vue';
+import UserSavedAddress from '~/components/UserSavedAddress';
+import SavedAddressForm from '~/components/MyAccount/SavedAddressForm';
+import { useUserShipping, userShippingGetters } from '@vue-storefront/spree';
+import { ref, computed } from '@nuxtjs/composition-api';
+import { onSSR } from '@vue-storefront/core';
+
+export default {
+  name: 'SavedAddressesDetails',
+  components: {
+    SfTabs,
+    SfButton,
+    SfIcon,
+    UserSavedAddress,
+    SavedAddressForm
+  },
+  setup() {
+    const { shipping, load: loadUserSavedAddresses, addAddress, deleteAddress, updateAddress } = useUserShipping();
+    const addresses = computed(() => userShippingGetters.getAddresses(shipping.value));
+    const edittingAddress = ref(false);
+    const activeAddress = ref(undefined);
+    const isNewAddress = computed(() => !activeAddress.value);
+
+    const changeAddress = (address = undefined) => {
+      activeAddress.value = address;
+      edittingAddress.value = true;
+    };
+
+    const removeAddress = address => deleteAddress({ address });
+
+    const saveAddress = async ({ form, onComplete, onError }) => {
+      try {
+        const actionMethod = isNewAddress.value ? addAddress : updateAddress;
+        const data = await actionMethod({ address: form });
+        edittingAddress.value = false;
+        activeAddress.value = undefined;
+        await onComplete(data);
+      } catch (error) {
+        onError(error);
+      }
+    };
+
+    onSSR(async () => {
+      await loadUserSavedAddresses();
+    });
+
+    return {
+      changeAddress,
+      updateAddress,
+      removeAddress,
+      saveAddress,
+      userShippingGetters,
+      addresses,
+      edittingAddress,
+      activeAddress,
+      isNewAddress
+    };
+  }
+};
+</script>
+
+<style lang='scss' scoped>
+
+.message {
+  font-family: var(--font-family--primary);
+  line-height: 1.6;
+  font-size: var(--font-size--base);
+  margin: 0 0 var(--spacer-base);
+}
+.savedAddresses-list {
```

Comment:
Overall the codebase uses BEM convention for naming css classes, so this would be more like `saved-addresses__list`

Reply:
Fixed! It really couldn't be saved-addresses__list because of the missing saved-addresses parent element. Hence, I've updated the _SavedAddressesDetails.vue_ by refactoring the HTML/CSS structure to fit the BEM convention by moving the saved-addresses class higher in the hierarchy of the elements.

Reply:
@letelete I think there's still a leftover here https://github.com/vuestorefront/spree/pull/156/files#diff-37a378edbefc1e450d4e225b207117be59a9e602e774373412fe0989afe6eb53R4